spec/git-repository-async-spec.js

```diff
@@ -0,0 +1,598 @@
+      repo.onDidChangeStatus(statusHandler)
+      fs.writeFileSync(filePath, '')
+
+      await repo.getPathStatus(filePath)
+
+      expect(statusHandler.callCount).toBe(1)
+      const status = Git.Status.STATUS.WT_MODIFIED
+      expect(statusHandler.argsForCall[0][0]).toEqual({path: filePath, pathStatus: status})
+      fs.writeFileSync(filePath, 'abc')
+
+      await repo.getPathStatus(filePath)
+      expect(statusHandler.callCount).toBe(1)
+    })
+  })
+
+  describe('.getDirectoryStatus(path)', () => {
+    let directoryPath, filePath
+
+    beforeEach(() => {
+      const workingDirectory = copyRepository()
+      repo = GitRepositoryAsync.open(workingDirectory)
+      directoryPath = path.join(workingDirectory, 'dir')
+      filePath = path.join(directoryPath, 'b.txt')
+    })
+
+    it('gets the status based on the files inside the directory', async () => {
+      await repo.checkoutHead(filePath)
+
+      let result = await repo.getDirectoryStatus(directoryPath)
+      expect(repo.isStatusModified(result)).toBe(false)
+
+      fs.writeFileSync(filePath, 'abc')
+
+      result = await repo.getDirectoryStatus(directoryPath)
+      expect(repo.isStatusModified(result)).toBe(true)
+    })
+  })
+
+  describe('.refreshStatus()', () => {
+    let newPath, modifiedPath, cleanPath
+
+    beforeEach(() => {
+      const workingDirectory = copyRepository()
+      repo = GitRepositoryAsync.open(workingDirectory)
+      modifiedPath = path.join(workingDirectory, 'file.txt')
+      newPath = path.join(workingDirectory, 'untracked.txt')
+      cleanPath = path.join(workingDirectory, 'other.txt')
+      fs.writeFileSync(cleanPath, 'Full of text')
+      fs.writeFileSync(newPath, '')
+      newPath = fs.absolute(newPath) // specs could be running under symbol path.
+    })
+
+    it('returns status information for all new and modified files', async () => {
+      fs.writeFileSync(modifiedPath, 'making this path modified')
+      await repo.refreshStatus()
+
+      expect(await repo.getCachedPathStatus(cleanPath)).toBeUndefined()
+      expect(repo.isStatusNew(await repo.getCachedPathStatus(newPath))).toBeTruthy()
+      expect(repo.isStatusModified(await repo.getCachedPathStatus(modifiedPath))).toBeTruthy()
+    })
+  })
+
+  describe('buffer events', () => {
+    let repository
+
+    beforeEach(() => {
+      const workingDirectory = copyRepository()
+      atom.project.setPaths([workingDirectory])
+
+      // When the path is added to the project, the repository is refreshed. We
+      // need to wait for that to complete before the tests continue so that
+      // we're in a known state.
+      repository = atom.project.getRepositories()[0].async
+      waitsFor(() => !repository._isRefreshing())
+    })
+
+    it('emits a status-changed event when a buffer is saved', async () => {
+      const editor = await atom.workspace.open('other.txt')
+
+      editor.insertNewline()
+
+      let called
+      repository.onDidChangeStatus(c => called = c)
```

Comment:
I think it would be better to use a Jasmine spy here and wait on the `callCount` and then assert on  `toHaveBeenCalledWith` helper.